docs/content/preview/manage/ysql-major-upgrade.md

```diff
@@ -0,0 +1,190 @@
+
+## Before you begin
+
+{{< warning >}}
+v2.25 is a preview release that is only meant for evaluation purposes and should not be used in production.
+{{< /warning >}}
+
+- DDL statements are blocked for the duration of the upgrade. Consider executing all DDLs before the upgrade, and pause any jobs that might run DDLs.
+- Upgrade client drivers.
+
+    Upgrade all application client drivers to the new version. The client drivers are backwards compatible, and work with both the old and new versions of the database.
+- Your cluster must be running v2024.2.2.0 or later.
+
+    Deploy a new YugabyteDB cluster on version v2024.2.2.0 or later. If you have a pre-existing cluster, first upgrade it to the latest version in the v2024.2 series using the [upgrade instructions](../upgrade-deployment/).
+
+### Precheck
+
+New PostgreSQL major versions add many new features and performance improvements, but also remove some older unsupported features and data types. You can only upgrade after you remove all deprecated features and data types from your databases.
+
+Use the [pg_upgrade](https://www.postgresql.org/docs/15/pgupgrade.html) tool provided with v2.25.1.0 to make sure your cluster is compatible with v2.25.1.0.
+
+Use the `--check` option as follows:
+
+```sh
+pg_upgrade --check -U <superuser_name> --old-host <any_yb_node_address>  --old-port <port> --old-datadir <yb_data_dir>
+```
+
+For example:
+
+```sh
+$ ./yugabyte-2.25.1.0/postgres/bin/pg_upgrade --check -U yugabyte --old-host 127.0.0.1 --old-port 5433 --old-datadir ~/yugabyte-data/node-1/disk-1/pg_data_11
+```
+
+```output
+Performing Consistency Checks on Old Live Server
+------------------------------------------------
+Checking cluster versions                                   ok
+Checking attributes of the 'yugabyte' user                  ok
+Checking for all 3 system databases                         ok
+Checking database connection settings                       ok
+Checking for system-defined composite types in user tables  ok
+Checking for reg* data types in user tables                 ok
+Checking for user-defined postfix operators                 ok
+Checking for incompatible polymorphic functions             ok
+Checking for invalid "sql_identifier" user columns          ok
+
+*Clusters are compatible*
+```
+
+## Prepare the cluster for YSQL major upgrade
+
+Since the upgrade is fully online, your cluster will temporarily consist of a mix of nodes running both PostgreSQL 11 and 15 ("mixed mode"). These processes need to be able to talk with each other and correctly process your SQL commands. Specifically, the PostgreSQL expressions that are used in the SQL statements get pushed down from the compute layer to the YugabyteDB storage layer. Because the expressions used by PostgreSQL have changed across the major version, you need to disable this optimization during the upgrade. (This will be addressed in the v2025.1 ({{<issue 24730>}}).)
```

Comment:
```suggestion
Since the upgrade is fully online, your cluster will temporarily consist of a mix of nodes running both PostgreSQL 11 and 15 (mixed mode). These processes need to be able to talk with each other and correctly process your SQL commands. Specifically, the PostgreSQL expressions that are used in the SQL statements get pushed down from the compute layer to the YugabyteDB storage layer. Because the expressions used by PostgreSQL have changed across the major version, you need to disable this optimization during the upgrade. (This will be addressed in the v2025.1 ({{<issue 24730>}}).)
```